public/secure/login.js

```diff
@@ -57,28 +53,23 @@ function login(event) {
   const formData = new FormData(event.target)
   const data = new URLSearchParams(formData)
   const loginUrl = new URL('login_check', identityProviderUrl)
-  performLoginRequest(loginUrl, data)
-    .then(
-      () => 'login-processing',
-      (error) => {
-        if (error instanceof ForbiddenError) {
-          const fallbackUrl = new URL('login_check', fallbackIDP)
-          return performLoginRequest(fallbackUrl, data).then(
-            () => 'login-fallback',
-            () => 'login-fallback'
-          )
-        }
-
-        if (error instanceof UnauthorisedError) showWrongCredentialsMessage()
-        return 'login-processing'
+  performLoginRequest(loginUrl, data).then(
+    () => {
+      window.top.location.href = '/'
```

Comment:
Why `.top`?

This line breaks the architectural principle of self-containing. If it happens somehow that your parent app decides to host within some path, let's say `core.ac.uk/dashboard/`, this will break. Technically it should either accept a parameter or send a notification what it was before.

Reply:
> Why .top?

I found it on stackoverflow so I used it. There is no big difference betweenthese three I am aware of.
```
window.parent.window.location.href
parent.location.href
window.top
```


That's true but we would need to handle this everywhere (logout, actions...) not just here.